### ISSUE TITLE
Warning fix using MemberNullAttribute causes accessibility level error.

### ISSUE THREAD

**Issue by kvansaders** [RicoSuter/NSwag#4763]
> I think a better solution is to generate
> 
> ```cs
>  public string BaseUrl
>  {
>      get { return _baseUrl; }
>      [System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(_baseUrl))]
>      set
>      {
>          _baseUrl = value;
>          if (!string.IsNullOrEmpty(_baseUrl) && !_baseUrl.EndsWith("/"))
>              _baseUrl += '/';
>      }
>  }
> ```
> 
> To let the compiler null analysis know that the property setter sets the field _baseUrl.

In my netstandard2.0 project, this gives: 'MemberNotNullAttribute' is inaccessible due to its protection level.

_Originally posted by @kvansaders in https://github.com/RicoSuter/NSwag/issues/4704#issuecomment-1927882073_

**Comment by Vogel612** [comment]
I'm seeing the same issue as the [other commenter on that issue,](https://github.com/RicoSuter/NSwag/issues/4704#issuecomment-1929122191) where the build completely breaks with the following error message:

```text
Error	CS0234	The type or namespace name 'MemberNotNullAttribute' does not exist in the namespace 'System.Diagnostics.CodeAnalysis' (are you missing an assembly reference?)	
```

as it is, 14.0.3 breaks netstandard compat from what I can tell

**Comment by ImoutoChan** [comment]
Same issue on my side, MemberNotNull shouldn't be emitted when targeting netstandard2.0

**Comment by nessuarez** [comment]
The same issue for targeting WinX86 runtime for net framework 4.5.2

**Comment by jochenwezel** [comment]
The feature `System.Diagnostics.CodeAnalysis.MemberNotNull` introduced with .NET 5 as MS documentation declares at https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.membernotnullattribute?view=net-8.0

So, the code line 
```
            [System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(_baseUrl))]
```
should be as follows to solve the compilation error
```
#if NET5_0_OR_GREATER
            [System.Diagnostics.CodeAnalysis.MemberNotNull(nameof(_baseUrl))]
#endif
```

**Comment by rwb196884** [comment]
I need to multi-target `net40` and this makes 14 unusable, have to use 13.

**Comment by Vogel612** [comment]
Forward duplicated by #4833, fixed since 14.0.6
